src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -603,20 +617,19 @@ struct TypeChecker<'a, 'gcx: 'a + 'tcx, 'tcx: 'a> {
     region_bound_pairs: &'a [(ty::Region<'tcx>, GenericKind<'tcx>)],
     implicit_region_bound: Option<ty::Region<'tcx>>,
     reported_errors: FxHashSet<(Ty<'tcx>, Span)>,
-    constraints: MirTypeckRegionConstraints<'tcx>,
-    borrowck_context: Option<BorrowCheckContext<'a, 'tcx>>,
+    borrowck_context: &'a mut Option<BorrowCheckContext<'a, 'tcx>>,
```

Comment:
Seems odd for this to be `&mut Option<>` and not `Option<&mut>`, but it doesn't really matter. If you did the latter, you would have to do:

```rust
if let Some(borrowck_context) = &mut self.borrowck_context
```

instead of this

```rust
if let Some(borrowck_context) = self.borrowck_context
```

Reply:
Fixed this.